### PR TITLE
BUG: Use one calculator in ItermodesThresholder

### LINKED ISSUE #1720: Update modified time in `itk::IntermodesThresholdImageFilter::SetMaximumSmoothingIterations` and other set methods.
### Description

When calling the `Set...` methods in the  `itk::IntermodesThresholdImageFilter` class it should be update the classes MTime if the value changes. This will ensure the pipeline reexecutes after a value has changed.

### Steps to Reproduce
None.

### Expected behavior
Reproducing @blowekamp 's proposal in [this comment](https://github.com/InsightSoftwareConsortium/ITK/pull/1718/files#r396414648):

> The simple way to do this would be to copy the code in the [set/get macro](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/Core/Common/include/itkMacro.h#L908-L912) which checks if the value is the same, if they are different then set the modify time. The more complicated way would involve modify the parent's GetMTime method to return the max of the MTime of itself, and the m_Calculator.

### Actual behavior
No check about the modified time is made.

### Reproducibility
%100

### Versions
`master`

### Environment
Any environment.

### Additional Information
None.

### PR DESCRIPTION
**Pull request by blowekamp**
Use the parent class's calculator and override GenerateDate method to
update the derived calculators ivars. Store the calculator parameters
as filter ivar, with correct filter modified time management.

Closes #1720.

## PR Checklist
- [x] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [x] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)
- [x] Added test (or behavior not changed)
- [x] Updated API documentation (or API not changed)
- [ ] Added [license](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Utilities/KWStyle/ITKHeader.h) to new files (if any)
- [ ] Added Python wrapping to new files (if any) as described in [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) Section 9.5
- [ ] Added [ITK examples](https://github.com/InsightSoftwareConsortium/ITKExamples) for all new major features (if any)

Refer to the [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) for
further development details if necessary.

### CODE REVIEW

**Comment by blowekamp** [comment]
After reviewing the code, based on my understanding of the issue, and a couple iterations this is what I came up with to address a couple issue and keep the implementations perhaps a little easier to maintain.

**Comment by blowekamp** [comment]
The Printing of the new ivars is not done. The PrintSelf method should be overloaded.

File: Modules/Filtering/Thresholding/include/itkIntermodesThresholdImageFilter.h [hunk 1]
```diff
@@ -93,31 +93,15 @@ class IntermodesThresholdImageFilter : public HistogramThresholdImageFilter<TInp
...
-  {
-    m_IntermodesCalculator->SetMaximumSmoothingIterations(maxSmoothingIterations);
-  }
 
-  SizeValueType
-  GetMaximumSmoothingIterations()
-  {
-    return (m_IntermodesCalculator->GetMaximumSmoothingIterations());
-  }
+  itkSetMacro(UseInterMode, SizeValueType);
```
Comment (dzenanz) [inline Modules/Filtering/Thresholding/include/itkIntermodesThresholdImageFilter.h]: Should this be `MaximumSmoothingIterations`? `UseInterMode` is declared few lines later.

Comment (blowekamp) [reply Modules/Filtering/Thresholding/include/itkIntermodesThresholdImageFilter.h]: Good catch!

**Comment by jhlegarreta** [comment]
Could this PR be also used to take [this](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/Filtering/Thresholding/include/itkHistogramThresholdImageFilter.hxx#L84) and [this](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/Filtering/Thresholding/include/itkHistogramThresholdImageFilter.hxx#L97) statements outside the `if/else` block and thus use a single statement?

**Comment by blowekamp** [comment]
> Could this PR be also used to take [this](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/Filtering/Thresholding/include/itkHistogramThresholdImageFilter.hxx#L84) and [this](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/Filtering/Thresholding/include/itkHistogramThresholdImageFilter.hxx#L97) statements outside the `if/else` block and thus use a single statement?
> 
> Sorry Brad. Just realized that I had noticed this in my PR. Should have realized earlier.

I missed that difference. This PR is ready to be merged now, so can you create separate PR with that change? It's a little unrelated to this one.

**Comment by jhlegarreta** [comment]
> I missed that difference. This PR is ready to be merged now, so can you create separate PR with that change? It's a little unrelated to this one.

Sounds fair. Thanks Brad !!!

**Pull request merged**